### PR TITLE
cmd/istanbul, docker, genesis: add docker-compose and quorum flags

### PR DESCRIPTION
**Pull request by markya0616**

### CODE REVIEW

File: docker/compose/ibft.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	err = tmpl.Execute(result, bft)
+	if err != nil {
+		fmt.Printf("Failed to render template, %v", err)
+		return ""
+	}
+
+	return result.String()
+}
+
+var ibftTemplate = `version: '3'
```
Comment (tailingchen) [inline docker/compose/ibft.go]: How about making a template file instead of a string variable?

Comment (markya0616) [reply docker/compose/ibft.go]: Follow the coding style in go-ethereum. https://github.com/ethereum/go-ethereum/blob/master/accounts/abi/bind/template.go

File: README.md [hunk 2]
```diff
@@ -99,7 +99,7 @@ seal: 0x000000000000000000000000000000000000000000000000000000000000000000000000
 <details>
 <summary>Click here to expand</summary>
 
-When `--nodes --verbose` flags are given, a `static-nodes.json` template as well as the validators' node keys, public keys, and addresses are generated. When `--save` flag is given, the generated configs will be saved.
+When `--nodes --verbose` flags are given, a `static-nodes.json` template as well as the validators' node keys, public keys, and addresses are generated. When `--docker-compose` is given, a `docker-compose.yml` for the validators is generated. When `--save` flag is given, all generated configs will be saved. Use Quorum when `quorum` flag is given.
```
Comment (alanchchen) [inline README.md]: Use Quorum when `--quorum` flag is given.

File: genesis/genesis.go [hunk 3]
```diff
@@ -83,18 +83,24 @@ func NewFile(isQuorum bool, options ...Option) string {
...
-		return err
+		return err, ""
 	}
 
 	//Quorum hack: add isQuorum field
+	jsonStr := string(raw)
 	if isQuorum {
-		jsonStr := string(raw)
 		idx := strings.Index(jsonStr, ",\"istanbul\"")
 		jsonStr = fmt.Sprintf("%s,\"isQuorum\":true%s", jsonStr[:idx], jsonStr[idx:])
```
Comment (alanchchen) [inline genesis/genesis.go]: Try to define new `Genesis` and `ChainConfig` struct ?

File: genesis/genesis.go [hunk 4]
```diff
@@ -83,18 +83,24 @@ func NewFile(isQuorum bool, options ...Option) string {
...
 func Save(dataDir string, genesis *core.Genesis, isQuorum bool) error {
 	filePath := filepath.Join(dataDir, FileName)
+	err, jsonStr := ToString(genesis, isQuorum)
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(filePath, []byte(jsonStr), 0600)
+}
 
+func ToString(genesis *core.Genesis, isQuorum bool) (error, string) {
```
Comment (alanchchen) [inline genesis/genesis.go]: The return values should be `(string, error)`.

File: docker/service/eth_stats.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+	if err != nil {
+		fmt.Printf("Failed to render template, %v", err)
+		return ""
+	}
+
+	return result.String()
+}
+
+var ethStatsTemplate = `eth-stats:
+    image: quay.io/maicoin/eth-netstats:latest
```
Comment (alanchchen) [inline docker/service/eth_stats.go]: I suggest using amis repository

Comment (markya0616) [reply docker/service/eth_stats.go]: OK

File: genesis/genesis.go (hunk 4 above)
Comment (markya0616) [reply genesis/genesis.go]: OK

File: README.md (hunk 2 above)
Comment (markya0616) [reply README.md]: OK

File: genesis/genesis.go (hunk 3 above)
Comment (markya0616) [reply genesis/genesis.go]: OK

**Pull request merged**